tests/pytests/unit/utils/test_files.py

```diff
@@ -0,0 +1,157 @@
+"""
+Unit Tests for functions located in salt/utils/files.py
+"""
+
+
+import copy
+import os
+
+import pytest
+import salt.utils.files
+from tests.support.mock import patch
+
+
+def test_safe_rm():
+    with patch("os.remove") as os_remove_mock:
+        salt.utils.files.safe_rm("dummy_tgt")
+        assert os_remove_mock.called is True
+
+
+@pytest.mark.skipif(
+    os.path.exists("/tmp/no_way_this_is_a_file_nope.sh"),
+    reason="Test file exists! Skipping safe_rm_exceptions test!",
+)
+def test_safe_rm_exceptions():
+    error = False
+    try:
+        salt.utils.files.safe_rm("/tmp/no_way_this_is_a_file_nope.sh")
+    except OSError:
+        error = True
+    assert error is False
```

Comment:
```suggestion
def test_safe_rm_exceptions(tmp_path):
    with pytest.raises(OSError):
        salt.utils.files.safe_rm(str(tmp_path / "no_way_this_is_a_file_nope.sh"))
```
Even better, less code.

Reply:
@s0undt3ch the function `salt.utils.files.safe_rm()` isn't raising anything, I assumed it would based on the test but that's not the case.
```
def safe_rm(tgt):
    """
    Safely remove a file
    """
    try:
        os.remove(tgt)
    except OSError:
        pass
```

Reply:
I that case just do `assert salt.utils.files.safe_rm(str(tmp_path / "no_way_this_is_a_file_nope.sh")) is None`

Reply:
But it's pretty much a useless test because it always returns `None` wether it actually removed or not

Reply:
I'm in favor of removing it, but test coverage is never a bad thing...

Reply:
K, then assert against None, we'll know when that changes